docs/machine-learning/tutorials/time-series-demand-forecasting.md

```diff
@@ -0,0 +1,285 @@
+
+    [!code-csharp [SaveModel](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L52)]
+
+## Use the model to forecast demand
+
+1. Below the `Evaluate` method, create a new utility method called `Forecast`.
+
+    ```csharp
+    static void Forecast(IDataView testData, int horizon, TimeSeriesPredictionEngine<ModelInput, ModelOutput> forecaster, MLContext mlContext)
+    {
+
+    }
+    ```
+
+1. Inside the `Forecast` method, use the [`Predict`](xref:Microsoft.ML.Transforms.TimeSeries.TimeSeriesPredictionEngine%602.Predict*) method to forecast rentals for the next seven days.
+
+    [!code-csharp [SingleForecast](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L91)]
+
+1. Align the actual and forecast values for seven periods.
+
+    [!code-csharp [GetForecastOutput](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L93-L108)]
+
+1. Iterate through the forecast output and display it on the console.
+
+    [!code-csharp [DisplayForecast](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L111-L116)]
+
+## Run the application
+
+1. Inside the `Main` method, call the `Forecast` method.
+
+    [!code-csharp [BuildForecast](~/machinelearning-samples/samples/csharp/getting-started/Forecasting_BikeSharingDemand/BikeDemandForecasting/Program.cs#L54)]
+
+1. Run the application. Output similar to that below should appear on the console. For brevity, the output has been condensed.
+
+    ```text
+    Evaluation Metrics
+    ---------------------
+    Mean Absolute Error: 726.416
+    Root Mean Squared Error: 987.658
+
+    Rental Forecast
+    ---------------------
+    Date: 1/1/2012
+    Actual Rentals: 2294
+    Lower Estimate: 1197.842
+    Forecast: 2334.443
+    Upper Estimate: 3471.044
+
+    Date: 1/2/2012
+    Actual Rentals: 1951
+    Lower Estimate: 1148.412
+    Forecast: 2360.861
+    Upper Estimate: 3573.309
+    ```
+
+Inspection of the actual and forecasted values shows the following relationships:
+
+![Actual vs Forecast Comparison](./media/time-series-demand-forecasting/forecast.png)
+
+While the forecasted values are not predicting the exact number of rentals, they provide a more narrow range of values that allows an operation to optimize their use of resources. 
+
+Congratulations! You've now successfully built a time series machine learning model to forecast bike rental demand.
+
+You can find the source code for this tutorial at the [dotnet/machinelearning-samples](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/Forecasting_BikeSharingDemand) repository.
```

Comment:
```suggestion
You can find the source code for this tutorial at the [dotnet/machinelearning-samples](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/Forecasting_BikeSharingDemand) repository.

## Next steps

- [Machine learning tasks in ML.NET](../resources/tasks.md)
- [Improve model accuracy](../resources/improve-machine-learning-model-ml-net.md)

```

Reply:
A couple of cross-links that may be useful. We need to have x-links between articles.